src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -29,28 +30,55 @@ const RangeStep = styled.button`
   border: none;
   font-size: 0.8rem;
   cursor: pointer;
-  color: blue;
+  color: ${({ theme }) => theme.primary1};
   padding: 0;
 `
 
+const InvestBalance = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`
+
+const InvestMaxBalance = styled.button`
+  cursor: pointer;
+  color: ${({ theme }) => theme.primary1};
+  background: none;
+  border: none;
+`
+
 const INVESTMENT_STEPS = [0, 25, 50, 75, 100]
 
-export default function InvestOption({ approveData, updateInvestAmount, claim }: InvestOptionProps) {
+export default function InvestOption({ approveData, claim }: InvestOptionProps) {
   const { currencyAmount, price, cost: maxCost, investedAmount } = claim
+  const { updateInvestAmount } = useClaimDispatchers()
+
+  const [percentage, setPercentage] = useState<number>(0)
 
   const { account } = useActiveWeb3React()
 
   const token = currencyAmount?.currency
 
   const balance = useCurrencyBalance(account || undefined, token)
 
-  const handlePercentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    console.log(event.target.value)
-  }
+  const scaleValue = useCallback((number: number, inMin: number, inMax: number, outMin: number, outMax: number) => {
+    return ((number - inMin) * (outMax - outMin)) / (inMax - inMin) + outMin
```

Comment:
@nenadV91, as we discussed a few times, we cannot relay on basic floating operations. They have rounding errors.

also, a nit, avoid calling a variable number. Use some better name. Also number has a meaning in JS so also not good for that reason (you wouldn't call a variable boolean either).

Reply:
Updated function